docs/advanced/authentication.md

```diff
@@ -284,6 +284,45 @@ const willAuthError = ({ authState }) => {
 }
 ```
 
+This can be really useful when we know when our authentication state is invalid and want to prevent
+even sending any operation that we know will fail with an authentication error. However, if we were
+to use this and are logging in our users with a login _mutation_ then the above code will
+unfortunately never let this login mutation through to our GraphQL API.
+
+If we have such a mutation we may need to write a more sophisticated `willAuthError` function like
+the following:
+
+```js
+const willAuthError = ({ operation, authState }) => {
+  if (!authState) {
+    // Detect our login mutation and let this operation through:
+    return !(
+      operation.kind === 'mutation' &&
+      // Here we find any mutation definition with the "login" field
+      operation.query.definitions.some(definition => {
+        return (
+          definition.kind === 'OperationDefinition' &&
+          definition.selectionSet.selections.some(node => {
+            // The field name is just an example, since signup may also be an exception
+            return node.kind === 'Field' && node.name.value === 'login';
+          })
+        );
+      })
+    );
+
+    return true;
```

Comment:
We'd never actually reach this line. Could be worth assigning the above to a variable, e.g.
```js
const isLoginMutation = ... ;
return isLoginMutation ? false : true;
```